tests/functional/pyocf/types/cache.py

```diff
@@ -476,35 +516,76 @@ def generate_attach_config(
             _discard_on_start=False,
         )
 
-        return attach_cfg
+        self.write_lock()
 
+        c = OcfCompletion([("cache", c_void_p), ("priv", c_void_p), ("error", c_int)])
 
-    def attach_device(
-        self, device, force=False, perform_test=False, cache_line_size=None, open_cores=True
-    ):
-        attach_cfg = self.generate_attach_config(
-            device,
-            force,
-            perform_test,
-            cache_line_size,
-            open_cores=open_cores
+        self.owner.lib.ocf_mngt_cache_attach(
+            self.cache_handle, byref(attach_cfg), c, None
         )
+        c.wait()
+
+        self.write_unlock()
+
+        if c.results["error"]:
+            raise OcfError(
+                f"Attaching cache device failed",
+                c.results["error"],
+            )
 
+    def standby_attach(self, device, force=False):
         self.device = device
+        self.device_name = device.uuid
+
+        device_config = Cache.generate_device_config(device, perform_test=False)
+
+        attach_cfg = CacheAttachConfig(
+            _device=device_config,
+            _cache_line_size=self.cache_line_size,
+            _open_cores=False,
+            _force=force,
+            _discard_on_start=False,
+        )
 
         self.write_lock()
 
         c = OcfCompletion([("cache", c_void_p), ("priv", c_void_p), ("error", c_int)])
 
-        self.owner.lib.ocf_mngt_cache_attach(
+        self.owner.lib.ocf_mngt_cache_standby_attach(
             self.cache_handle, byref(attach_cfg), c, None
         )
-
         c.wait()
+
         self.write_unlock()
 
         if c.results["error"]:
-            raise OcfError("Attaching cache device failed", c.results["error"])
+            raise OcfError(
+                f"Attaching to standby cache failed",
+                c.results["error"],
+            )
+
+    def standby_load(self, device):
+        self.device = device
+        self.device_name = device.uuid
+
+        device_config = Cache.generate_device_config(device)
+
+        attach_cfg = CacheAttachConfig(
+            _device=device_config,
+            _cache_line_size=self.cache_line_size,
+            _open_cores=False,
+            _force=False,
+            _discard_on_start=False,
+        )
+
+        c = OcfCompletion([("cache", c_void_p), ("priv", c_void_p), ("error", c_int)])
+        self.owner.lib.ocf_mngt_cache_standby_load(
+            self.cache_handle, byref(attach_cfg), c, None
+        )
```

Comment:
We should take `self.write_lock()` for this operation. Regular `cache_load()` seems to be missing it as well.

Reply:
fixed